File: anul 2/semestrul 1/poo/kotlin/src/lab10/facade.kt
// Subsystem classes
class SubsystemA {
    fun operationA() {
        println("Subsystem A operation")
    }
}

class SubsystemB {
    fun operationB() {
        println("Subsystem B operation")
    }
}

class SubsystemC {
    fun operationC() {
        println("Subsystem C operation")
    }
}

// Facade class
class Facade {
    private val subsystemA = SubsystemA()
    private val subsystemB = SubsystemB()
    private val subsystemC = SubsystemC()

    fun complexOperation() {
        subsystemA.operationA()
        subsystemB.operationB()
        subsystemC.operationC()
    }
}

fun main() {
    val facade = Facade()
    facade.complexOperation()
}

File: anul 2/semestrul 1/poo/lab10/facade.cpp
#include <iostream>

// Subsystem classes
class SubsystemA {
public:
    void operationA() {
        std::cout << "Subsystem A operation" << std::endl;
    }
};

class SubsystemB {
public:
    void operationB() {
        std::cout << "Subsystem B operation" << std::endl;
    }
};

class SubsystemC {
public:
    void operationC() {
        std::cout << "Subsystem C operation" << std::endl;
    }
};

// Facade class
class Facade {
private:
    SubsystemA subsystemA;
    SubsystemB subsystemB;
    SubsystemC subsystemC;

public:
    void complexOperation() {
        subsystemA.operationA();
        subsystemB.operationB();
        subsystemC.operationC();
    }
};

int main() {
    Facade facade;
    facade.complexOperation();

    return 0;
}
